### PR TITLE
added updater page

### PR DESCRIPTION
**Pull request by 0PandaDEV**
#### What kind of changes does this PR include?
- added updater page

#### Description
Most of the content is inspired by the old [v1 version](https://tauri.app/v1/guides/distribution/updater) but with updated methods and more clearer instructions.

- [x] Checked Grammar spelling

### CODE REVIEW

File: src/content/docs/plugin/updater.mdx [hunk 1]
```diff
@@ -9,9 +9,285 @@ sidebar:
...
+  ```zsh
+  bunx tauri signer generate -w ~/.tauri/myapp.key
+  ```
+  </TabItem>
+  <TabItem label="cargo">
+  ```zsh
+  cargo tauri signer generate -w ~/.tauri/myapp.key
+  ```
+  </TabItem>
+</Tabs>
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]: could you replace these with command tabs, you can place bun too, ~right now the component doesn't have support but there's a PR to fix that.~

File: src/content/docs/plugin/updater.mdx [hunk 2]
```diff
@@ -9,9 +9,285 @@ sidebar:
...
+  ```cmd
+  bunx tauri signer generate -w $HOME/.tauri/myapp.key
+  ```
+  </TabItem>
+  <TabItem label="cargo">
+  ```cmd
+  cargo tauri signer generate -w $HOME/.tauri/myapp.key
+  ```
+  </TabItem>
+</Tabs>
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]: same

File: src/content/docs/plugin/updater.mdx [hunk 3]
```diff
@@ -9,9 +9,285 @@ sidebar:
...
+
+### Building
+
+While building your update artifacts, you need to have the private key in your ENVIRONMENT variables.
+
+<Tabs>
+  <TabItem label="Mac/Linux">
+  ```zsh
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]:
```suggestion
  ```sh frame=none
```
we usually use ```sh

File: src/content/docs/plugin/updater.mdx [hunk 4]
```diff
@@ -9,9 +9,285 @@ sidebar:
...
+  export TAURI_SIGNING_PRIVATE_KEY=""
+  # optionally also add a password
+  export TAURI_KEY_PASSWORD=""
+  ```
+  </TabItem>
+  <TabItem label="Windows">
+  Run this in `PowerShell`:
+  ```cmd
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]:
```suggestion
  ```ps frame=none
```
and ps for windows commands

File: src/content/docs/plugin/updater.mdx [hunk 5]
```diff
@@ -9,9 +9,285 @@ sidebar:
...
+| `pubkey`       | This has to be the public key generated from the Tauri CLI in the step above.          |
+| `endpoints`    | This must be an array of endpoint URLs as strings. TLS is enforced in production mode.  |
+
+Each updater URL can contain the following dynamic variables, allowing you to determine server-side if an update is available.
+
+- `{{current_version}}`: The version of the app that is requesting the update.
+- `{{target}}`: The operating system name (one of Linux, windows or Darwin).
+- `{{arch}}`: The architecture of the machine (one of x86_64, i686, aarch64 or armv7).
+
+```json
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]:
```suggestion
```json title=tauri.conf.json
```

**Review by vasfvitor — CHANGES_REQUESTED** [review]
hey, thank you for this, content will be reviewed soon, for now a few changes to the code

File: src/content/docs/plugin/updater.mdx [hunk 6]
```diff
@@ -9,9 +9,249 @@ sidebar:
...
+
+<Tabs>
+  <TabItem label="Static">
+
+  When using static, you just need to return a JSON containing the required information.
+
+  | Keys           | Description                                                                                                                                 |
+  | -------------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+  | `version`      | Must be a valid [SemVer](https://semver.org/), with or without a leading v, meaning that both 1.0.0 and v1.0.0 are valid.                   |
+  | `notes`        | Notes about the update. This will be displayed in the dialog Tauri generates.                                                               |
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]:
```suggestion
  | `notes`        | Notes about the update.                                                  |
```

Comment (vasfvitor) [reply src/content/docs/plugin/updater.mdx]: in v2 there's no default dialog anymore, one have to manually create if wanted

File: src/content/docs/plugin/updater.mdx [hunk 7]
```diff
@@ -9,9 +9,249 @@ sidebar:
...
+
+  Your server should respond with a status code of [`204 No Content`](https://datatracker.ietf.org/doc/html/rfc2616#section-10.2.5) if there is no update available.
+
+  If an update is required, your server should respond with a status code of [`200 OK`](http://tools.ietf.org/html/rfc2616#section-10.2.1) and a JSON response in this format:
+
+  | Keys           | Description                                                                                                                                 |
+  | -------------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+  | `version`      | This Must be a valid [SemVer](https://semver.org/), with or without a leading v, meaning that both 1.0.0 and v1.0.0 are valid.              |
+  | `notes`        | Notes about the update. This will be displayed in the dialog Tauri generates.                                                               |
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]:
```suggestion
  | `notes`        | Notes about the update.                                                               |
```

File: src/content/docs/plugin/updater.mdx [hunk 8]
```diff
@@ -9,9 +9,249 @@ sidebar:
...
+  $env:TAURI_SIGNING_PRIVATE_KEY=""
+  <# optionally also add a password #>
+  $env:TAURI_KEY_PASSWORD=""
+  ```
+  </TabItem>
+</Tabs>
+
+After that, you can run Tauri build as usual and Tauri will generate the update bundle and its signature.
+
+- Linux: On Linux, Tauri will create a .tar.gz archive from the AppImage inside the target/release/bundle/appimage/ folder:
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]:
```suggestion
- Linux: On Linux, Tauri will create a .tar.gz archive from the AppImage inside the `target/release/bundle/appimage/` folder:  
```

same on others path on this page

File: src/content/docs/plugin/updater.mdx [hunk 9]
```diff
@@ -9,9 +9,249 @@ sidebar:
...
+  # optionally also add a password
+  export TAURI_KEY_PASSWORD=""
+  ```
+  </TabItem>
+  <TabItem label="Windows">
+  Run this in `PowerShell`:
+  ```ps frame=none
+  $env:TAURI_SIGNING_PRIVATE_KEY=""
+  <# optionally also add a password #>
+  $env:TAURI_KEY_PASSWORD=""
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]:
```suggestion
  $env:TAURI_SIGNING_PRIVATE_KEY_PASSWORD=""
```

https://beta.tauri.app/start/migrate/from-tauri-1/#environment-variables-changes

File: src/content/docs/plugin/updater.mdx [hunk 10]
```diff
@@ -9,9 +9,249 @@ sidebar:
...
+### Building
+
+While building your update artifacts, you need to have the private key in your ENVIRONMENT variables.
+
+<Tabs>
+  <TabItem label="Mac/Linux">
+  ```sh frame=none
+  export TAURI_SIGNING_PRIVATE_KEY=""
+  # optionally also add a password
+  export TAURI_KEY_PASSWORD=""
```
Comment (vasfvitor) [inline src/content/docs/plugin/updater.mdx]:
```suggestion
  export TAURI_SIGNING_PRIVATE_KEY_PASSWORD=""
```

**Review by vasfvitor — DISMISSED** [review]
Could you add 2 TODO comments on the top of the file, one for adding CrabNebula tip about updater and another to add a tip about GitHub action

Still missing the api usage, both Rust and Javascript, it can be as simples as this:

https://beta.tauri.app/start/migrate/from-tauri-1/#migrate-to-updater-plugin

https://github.com/tauri-apps/tauri-plugin-updater?tab=readme-ov-file#usage

And the permission list "updater:default", I'm not sure Tauri adds it by default, but it's important to list all the permissions from this table here:

https://github.com/tauri-apps/tauri-plugin-updater/blob/v2/permissions/autogenerated/reference.md

You can check other plugin guides to see how they done it so you can get an idea, for instance:

https://beta.tauri.app/plugin/barcode-scanner/#usage
https://beta.tauri.app/plugin/barcode-scanner/#permissions

Permission can go in the end of the guide. Usage I'm not so sure, it could go after Setup or before permissions, see what do you think it's best for someone trying it out.

And I think this does it, let me know if you have any ideas

**Comment by FabianLars** [comment]
Let's worry about js/rust/permissions later. Thanks for the contribution!

**Review by FabianLars — APPROVED** [review]

**Pull request merged**